torch/distributed/distributed_c10d.py

```diff
@@ -4297,6 +4298,175 @@ def _get_backend_from_str(backend: Optional[str] = None) -> Backend:
     return Backend(backend)
 
 
+@_time_logger
+def split_group(
```

Comment:
please add mypy typing hints

Reply:
myspy tying hints does not work well with 'None' default value, e.g.,
default has type "None", argument has type "timedelta"

Reply:
it takes some work to get right but i think its usually worthwhile.  In general you can use Optional[] or Union[] to make the None case or the multi-type cases work out.  Then it'll force you to assert its not none or assert its the right type before using it.

Reply:
just double checked, other APIs in the file are using mypy.  if init_process_group can do it then you can too :) 

```
@_time_logger
def init_process_group(
    backend: Optional[str] = None,
    init_method: Optional[str] = None,
    timeout: Optional[timedelta] = None,
    world_size: int = -1,
    rank: int = -1,
    store: Optional[Store] = None,
    group_name: str = "",
    pg_options: Optional[Any] = None,
    device_id: Optional[torch.device] = None,
) -> None:
```